### PR TITLE
Assign monitor to pool

### PR DESCRIPTION
**Pull request by chrisUsick**
The feature allows you to add a monitor to a pool. The changes I made to the Vagrantfile and the gemfile were to get the repo to run with the latest version of vagrant.

### CODE REVIEW

File: resources/pool.rb [hunk 1]
```diff
@@ -5,5 +5,6 @@
 attribute :host, :kind_of => String, :regex => /.*/
 attribute :ip, :kind_of => String, :regex => /.*/
 attribute :port, :kind_of => Integer, :regex => /^\d+$/
+attribute :monitor, :kind_of => String, :regex => /.*$/
```
Comment (swalberg) [inline resources/pool.rb]: Just being pedantic here, but the `$` is redundant.

File: libraries/chef_f5.rb [hunk 2]
```diff
@@ -41,6 +44,33 @@ def pool_is_missing_node?(pool, node)
...
+    api.LocalLB.Pool.set_monitor_association(monitor_associations: {
+      item: [
+        {pool_name: pool,
+         monitor_rule: {
+           monitor_templates: {
+             item: monitor
+           }
+         },
+         quorum: '0',
+         type: 'MONITOR_RULE_TYPE_SINGLE'
```
Comment (swalberg) [inline libraries/chef_f5.rb]: Hm I hadn't considered multiple monitors. Does this break if the pool already has another monitor?

I don't think this PR needs to address the more complicated scenarios but I just want to make sure it's not going to break existing stuff.

File: providers/pool.rb [hunk 3]
```diff
@@ -40,4 +40,21 @@ def whyrun_supported?
...
+
+  if new_resource.monitor
+    if f5.pool_is_missing_monitor?(new_resource.name, new_resource.monitor)
+      converge_by("Add monitor #{new_resource.monitor} to pool #{new_resource.name}") do
+        begin
+          f5.add_monitor(new_resource.name, new_resource.monitor)
+          new_resource.updated_by_last_action(true)
+          Chef::Log.info("#{new_resource} added monitor #{new_resource.monitor} to pool #{new_resource.name}")
+        rescue
+          Chef::Log.warn("Adding monitor #{new_resource.monitor} failed. Ensure it exists.")
```
Comment (swalberg) [inline providers/pool.rb]: :+1:

File: providers/pool.rb [hunk 4]
```diff
@@ -40,4 +40,21 @@ def whyrun_supported?
...
+        end
+
+      end
+    else
+      Chef::Log.warn("#{new_resource.name} already has monitor #{new_resource.monitor}")
```
Comment (swalberg) [inline providers/pool.rb]: Could you please change this to log at `info`?

**Comment by swalberg** [comment]
Awesome, thanks for this fix. Couple of small things to talk about. Thanks again!

File: libraries/chef_f5.rb (hunk 2 above)
Comment (chrisUsick) [inline libraries/chef_f5.rb]: What `set_monitor_association` does is remove all monitors, and creates a single association to the given monitor. If the parameter `monitor` is an array of monitors, then a monitor association is created with an array of monitor templates, and the type is automatically set to `MONITOR_RULE_TYPE_AND_LIST`, regardless of the type passed in. A pool can have only 1 monitor association. Therefore, if `monitor` is an array and for each item in `monitor` you created a new monitor association only the last one is used. I confirmed all this by actually testing it. To sum it up: **this method sets the monitor, not adds a monitor**. To _add_ a monitor you would just have to query iControl to get the current monitors and append the new monitor to the current monitors in the call to `set_monitor_association`

**Pull request merged**